crates/c-api/src/store.rs

```diff
@@ -203,6 +204,28 @@ pub extern "C" fn wasmtime_context_set_wasi(
     })
 }
 
+#[cfg(feature = "wasi")]
+#[no_mangle]
+pub extern "C" fn wasmtime_context_set_default_wasi_if_not_exist(
+    mut context: CStoreContextMut<'_>,
+) -> bool {
+    let wasi = WasiCtxBuilder::new().build();
+
+    match context.data_mut().wasi.as_mut() {
+        Some(_) => false,
+        None => {
+            context.data_mut().wasi = Some(wasi);
+            true
+        }
+    }
+}
+
+#[cfg(feature = "wasi")]
+#[no_mangle]
+pub extern "C" fn wasmtime_context_get_wasi_ctx(context: CStoreContext<'_>) -> Box<WasiCtx> {
+    Box::new(context.data().wasi.as_ref().unwrap().clone())
```

Comment:
It is true, but are there any reason that the pointer would change, without calling SetWasi? 

SetWasi is apparently something a caller shouldn't call, since I would assume all previous states saved in the older wasi will be overridden anyways.

Reply:
Yes this is where I would recommend `wasmtime_context_*` methods instead of a separate `wasi_ctx_t` type. That would operate directly on a store and avoid the need to pull out a WASI pointer.

Reply:
Yeah I would love to move this API to `wasmtime_context_*` if you see it fit, but perhaps it would cause misalignment between Rust, C-API, and caller (say Go). Exposing the same function on the same struct here seems a much straightforward approach, maybe even with some memory safety trade-offs. 

Another way we could do is, exposing it here as `wasmtime_context_*` and implement a "fake" type `WasiCtx` in Go...

Reply:
Removed this function along with `wasmtime_context_set_default_wasi_if_not_exist`, added `wasmtime_context_insert_file` and `wasmtime_context_push_file` which directly call into the `WasiCtx` deep inside a `CStoreContextMut`.